rules/examples/typed_rule.py

```diff
@@ -0,0 +1,131 @@
+# This is an example of a rule with explicit types.  The base Rule class is typed,
+# so if you inherit from it, explicit typing is optional.  However, if you want to
+# be explicit about the types, you can do so like this:
+
+from typing import Dict, List
+
+from panther_core.enriched_event import PantherEvent
+from pydantic import NonNegativeInt, PositiveInt
+from pypanther import LogType, Rule, RuleTest, Severity
+from pypanther.base import SeverityType
+from pypanther.helpers.base import aws_guardduty_context
+from pypanther.severity import SEVERITY_DEFAULT
+from time import strptime
+
+
+class MyTypedRule(Rule):
+    log_types: List[LogType | str] = [LogType.AWS_GUARDDUTY]
+    id: str = "AWS.GuardDuty.HighVolFindings"
```

Comment:
Maybe edit the `id` to name is appropriately to the class (so people aren't confused if accidentally registering)